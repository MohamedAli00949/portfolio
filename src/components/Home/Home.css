.home-container {
  color: rgb(255, 255, 255);
  position: absolute;
  top: 50%;
  transform: translate(0px, -30%);
  display: flex;
  flex-flow: column nowrap;
  gap: 20px;
  place-content: center;
  align-items: center;
}

.home-container p:first-of-type {
  font-size: 93px;
  margin-block: 20px 0px;
  color: rgb(106, 223, 255);
  text-shadow: rgba(0, 0, 0, 0.37) 2px -1px 0px;
  text-transform: uppercase;
  font-weight: 700;
}

.home-container h1 {
  margin-block: 10px;
  font-size: 80px;
}

.home-container .logo {
  font-size: 82px;
  margin-inline: 0;
}

.home-container p.details {
  font-size: 28px;
  margin-inline: 10px;
  margin-block: 0px;
}

.social-media {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

a.social-icon {
  font-size: 35px;
  margin-inline: 10px;
  color: rgb(255, 255, 255);
  padding: 0 6.5px;
  margin: 0;
}

a.social-icon:hover {
  color: #09c6f9;
}

a.facebook:hover {
  color: #1b74e4;
}

a.linkedin:hover {
  color: #0a66c2;
}

#arrow {
  transition: all;
  position: absolute;
  top: 75px;
  animation-name: arrow;
  animation-duration: 1s;
  animation-delay: 2s;
  animation-iteration-count: infinite;
  animation-direction: alternate;
}

@keyframes arrow {
  0% {
    top: 0px;
  }
  100% {
    top: 60px;
  }
}

/* start responsive design */
@media screen and (max-width: 900px) {
  .home-container p:first-of-type {
    margin: 0;
  }

  .home-container h1 {
    margin: 0;
  }

  @keyframes arrow {
    0% {
      top: 40px;
    }

    100% {
      top: 120px;
    }
  }
}

@media screen and (max-width: 768px) {
  .home-container {
    gap: 25px;
  }

  .home-container p:first-of-type {
    font-size: 93px;
  }

  @keyframes arrow {
    0% {
      top: 0;
    }

    100% {
      top: 125px;
    }
  }
}

@media screen and (max-width: 730px) {
  @keyframes arrow {
    0% {
      top: 0;
    }

    100% {
      top: 70px;
    }
  }
}

@media screen and (max-width: 590px) {
  .home-container {
    gap: 20px;
  }

  .home-container p:first-of-type {
    font-size: 65px;
  }

  .home-container h1,
  .home-container .logo {
    font-size: 72px;
    margin-inline: 10px;
  }
}

@media screen and (max-width: 425px) {
  .home-container p:first-of-type {
    font-size: 93px;
  }

  @keyframes arrow {
    0% {
      top: 0;
    }

    100% {
      top: 75px;
    }
  }
}

@media screen and (max-width: 320px) {
  .home-container p:first-of-type {
    font-size: 44px;
  }

  .home-container h1,
  .home-container .logo {
    font-size: 40px;
  }

  .home-container p.details {
    margin-block: 0px;
    font-size: 22px;
    margin-inline: 10px;
  }
}
/* End Responisve Design */
