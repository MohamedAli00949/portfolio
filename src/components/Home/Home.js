import React, { lazy } from "react";
import Image from "../../images/james-wheeler-AFC0XvICMgs-unsplash.webp";

import { MdOutlineDoubleArrow } from "react-icons/md";
const Navbar = lazy(() => import("./../Navbar/Navbar"));
const Container = lazy(() => import("@mui/material/Container"));
const Grow = lazy(() => import("@mui/material/Grow"));
const Link = lazy(() => import("@mui/material/Link"));

const Home = () => {
  return (
    <div
    className="home"
      style={{
        backgroundImage: `url(${Image})`,
        width: "100%",
        height: "100vh",
        display: "flex",
        flexDirection: "column",
      }}
    >
      <Navbar />
      <section id="Home" className="section-container">
        <Container
          fixed
          style={{ textAlign: "center", margin: "auto", width: "100%" }}
        >
          <Grow
            direction="down"
            in
            style={{
              textAlign: "center",
              display: "flex",
              flexDirection: "column",
              justifyContent: "center",
            }}
          >
            <div
              style={{
                margin: "auto",
                display: "flex",
                alignItems: "center",
                justifyContent: "space-around",
                height: "max-content",
              }}
            >
              <div className="home-container">
                <p>Hey</p>
                <h1>
                  I am <span className="logo">Mohamed Ali Fawzi</span>
                </h1>
                <p className="details">
                  I am{" "}
                  <strong style={{ color: "rgb(106, 223, 255)" }}>
                    Software Engineer
                  </strong>{" "}
                  from Egypt.
                  <br />I am{" "}
                  <strong style={{ color: "rgb(106, 223, 255)" }}>
                    Computer Science Student
                  </strong>
                  .
                </p>
                <div style={{ position: "relative" }}>
                  <Link
                    aria-label="about arrow"
                    href="#About"
                    onClick={() => {
                      document
                        .querySelector(".section-link.active")
                        .classList.remove("active");
                      document
                        .getElementsByClassName("About")[0]
                        .classList.add("active");
                    }}
                  >
                    <MdOutlineDoubleArrow
                      id="arrow"
                      style={{
                        transform: "rotate(90deg)",
                        color: "#09c6f9",
                        fontSize: "30px",
                      }}
                    />
                  </Link>
                </div>
              </div>
            </div>
          </Grow>
        </Container>
      </section>
    </div>
  );
};

export default Home;
