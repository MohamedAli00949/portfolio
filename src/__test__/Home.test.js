import React, { Suspense } from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import Home from "../components/Home/Home";

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    container.setAttribute("id", "containeer")
    document.body.appendChild(container);
});

afterEach(() => {
    ReactDOM.unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("render footer section", async () => {
    await act(async () => {
        await render(<Suspense><Home /></Suspense>)
    });
    
    container = await document.body;

    expect(container.querySelector("section").getAttribute("id")).toBe("Home");
    expect(container.querySelector(".navbar").classList.contains("navbar")).toBeTruthy();
    
    expect(container.querySelector(".home-container p").textContent).toBe("Hey");
    expect(container.querySelector(".home-container h1").textContent).toBe("I am Mohamed Ali Fawzi");
    expect(container.querySelector(".home-container p.details").textContent).toBe("I am Software Engineer from Egypt.I am Computer Science Student.");
    expect(container.querySelector(".home-container div a").getAttribute("href")).toBe("#About");

    expect(container).toMatchSnapshot();
})