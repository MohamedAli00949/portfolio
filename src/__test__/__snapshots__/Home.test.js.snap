// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render footer section 1`] = `
<body>
  <div
    id="containeer"
  />
  <div>
    <div
      class="home"
      style="background-image: url(james-wheeler-AFC0XvICMgs-unsplash.webp); width: 100%; height: 100vh; display: flex; flex-direction: column;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed navbar mui-fixed css-10srjip-MuiPaper-root-MuiAppBar-root"
          style="width: 100%; padding: 5px 20px; color: black; flex-direction: column; align-items: center; justify-content: space-between; box-shadow: 0 0 0; top: 0px;"
        >
          <div
            class="nav-container"
            style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
          >
            <h1
              style="margin: 0px;"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways logo css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="/"
                style="text-decoration: none; color: rgb(9, 198, 249);"
              >
                MAF
              </a>
            </h1>
            <ul
              class="nav-section"
            >
              <li>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link active Home css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="#Home"
                  style="text-decoration: none;"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false About css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="#About"
                  style="text-decoration: none;"
                >
                  About
                </a>
              </li>
              <li>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false Portfolio css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="#Portfolio"
                  style="text-decoration: none;"
                >
                  Portfolio
                </a>
              </li>
              <li>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false Skills css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="#Skills"
                  style="text-decoration: none;"
                >
                  Skills
                </a>
              </li>
              <li>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways section-link false Contact css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="#Contact"
                  style="text-decoration: none;"
                >
                  Contact
                </a>
              </li>
            </ul>
            <div>
              <button
                aria-label="navMenu"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium nav-menu css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  color="white"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: white;"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </header>
      </div>
      <section
        class="section-container"
        id="Home"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
          style="text-align: center; margin: auto; width: 100%;"
        >
          <div
            direction="down"
            style="opacity: 1; transform: none; text-align: center; display: flex; flex-direction: column; justify-content: space-around; margin: auto; align-items: center; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <div
              class="home-container"
            >
              <p>
                Hey
              </p>
              <h1>
                I am 
                <span
                  class="logo"
                >
                  Mohamed Ali Fawzi
                </span>
              </h1>
              <p
                class="details"
              >
                I am
                 
                <strong
                  style="color: rgb(106, 223, 255);"
                >
                  Software Engineer
                </strong>
                 
                from Egypt.
                <br />
                I am
                 
                <strong
                  style="color: rgb(106, 223, 255);"
                >
                  Computer Science Student
                </strong>
                .
              </p>
              <div
                style="position: relative;"
              >
                <a
                  aria-label="about arrow"
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="#About"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    id="arrow"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(9, 198, 249); transform: rotate(90deg); font-size: 30px;"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M15.5 5H11l5 7-5 7h4.5l5-7z"
                    />
                    <path
                      d="M8.5 5H4l5 7-5 7h4.5l5-7z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</body>
`;
